src/wallet/wallet2.cpp

```diff
@@ -507,15 +507,18 @@ std::unique_ptr<wallet2> wallet2::make_dummy(const boost::program_options::varia
 }
 
 //----------------------------------------------------------------------------------------------------
-bool wallet2::init(std::string daemon_address, boost::optional<epee::net_utils::http::login> daemon_login, uint64_t upper_transaction_size_limit)
+bool wallet2::init(std::string daemon_address, boost::optional<epee::net_utils::http::login> daemon_login, uint64_t upper_transaction_size_limit, bool ssl)
 {
-  if(m_http_client.is_connected())
-    m_http_client.disconnect();
+    
   m_is_initialized = true;
   m_upper_transaction_size_limit = upper_transaction_size_limit;
   m_daemon_address = std::move(daemon_address);
   m_daemon_login = std::move(daemon_login);
-  return m_http_client.set_server(get_daemon_address(), get_daemon_login());
+
+  if(m_http_client.is_connected())
+    m_http_client.disconnect();
+  
```

Comment:
Only possible glitch I can think of is that client won't disconnect if the address can't be parsed (assert in set_server fails).